src/Core/Checkout/Cart/SalesChannel/CartOrderRoute.php

```diff
@@ -66,65 +70,75 @@ public function order(Cart $cart, SalesChannelContext $context, RequestDataBag $
             throw CartException::hashMismatch($cart->getToken());
         }
 
-        // we use this state in stock updater class, to prevent duplicate available stock updates
-        $context->addState('checkout-order-route');
-
-        $calculatedCart = $this->cartCalculator->calculate($cart, $context);
-
-        $response = $this->checkoutGatewayRoute->load(new Request($data->all(), $data->all()), $cart, $context);
-        $calculatedCart->addErrors(...$response->getErrors());
-
-        $this->taxProviderProcessor->process($calculatedCart, $context);
-
-        $this->addCustomerComment($calculatedCart, $data);
-        $this->addAffiliateTracking($calculatedCart, $data);
-
-        $preOrderPayment = Profiler::trace('checkout-order::pre-payment', fn () => $this->paymentProcessor->validate($calculatedCart, $data, $context));
-
-        $orderId = Profiler::trace('checkout-order::order-persist', fn () => $this->orderPersister->persist($calculatedCart, $context));
-
-        if (Feature::isActive('v6.8.0.0')) {
-            $this->cartPersister->delete($context->getToken(), $context);
-        }
-
-        $criteria = new Criteria([$orderId]);
-        $criteria
-            ->setTitle('order-route::order-loading')
-            ->addAssociation('orderCustomer.customer')
-            ->addAssociation('orderCustomer.salutation')
-            ->addAssociation('deliveries.shippingMethod')
-            ->addAssociation('deliveries.shippingOrderAddress.country')
-            ->addAssociation('deliveries.shippingOrderAddress.countryState')
-            ->addAssociation('transactions.paymentMethod')
-            ->addAssociation('lineItems.cover')
-            ->addAssociation('lineItems.downloads.media')
-            ->addAssociation('currency')
-            ->addAssociation('addresses.country')
-            ->addAssociation('addresses.countryState')
-            ->addAssociation('stateMachineState')
-            ->addAssociation('deliveries.stateMachineState')
-            ->addAssociation('transactions.stateMachineState')
-            ->getAssociation('transactions')->addSorting(new FieldSorting('createdAt'));
-
-        $this->eventDispatcher->dispatch(new CheckoutOrderPlacedCriteriaEvent($criteria, $context));
-
-        $orderEntity = Profiler::trace('checkout-order::order-loading', function () use ($criteria, $context): ?OrderEntity {
-            return $this->orderRepository->search($criteria, $context->getContext())->getEntities()->first();
-        });
-
-        if (!$orderEntity) {
-            throw CartException::invalidPaymentOrderNotStored($orderId);
+        $lock = $this->lockFactory->createLock('cart-order-route-' . $cart->getToken(), self::LOCK_TTL);
```

Comment:
just an idea. would it be somehow possible to have this on every cart route? :thinking: 

a few weeks ago @patzick came up with this issue: https://github.com/shopware/shopware/issues/8724

or would we need to implement that in every route?

Reply:
thank you! right, they are slightly different :+1: